src/cls/IPM/General/LockFile.cls

```diff
@@ -0,0 +1,125 @@
+/// This class is used for interacting with a lock file for a given module
+/// Covers functionality for both creating a lock file and installing from a lock file
```

Comment:
With regards to separation of concerns, it may be better for this class to just be the interface for interacting with the lock file and have the creation/install methods in a utils class?